source/slang/slang-ast-type.h

```diff
@@ -107,6 +107,19 @@ class BuiltinType : public DeclRefType
     SLANG_ABSTRACT_CLASS(BuiltinType)
 };
 
+class FeedbackType : public BuiltinType
+{
+    SLANG_CLASS(FeedbackType)
+
+    enum class Kind : uint8_t
```

Comment:
I'm always a bit uneasy using "kind" in a way that doesn't really refer to the "type of a type."

AFAICT, in this case the `Kind` enumeration is used to tell which of two concrete types we have, such that the information in `kind` is redundant with the `DeclRef` that the type refers to. Is it possible to just remove this field entirely?

Reply:
Having looked through the rest of the PR, I'm not seeing this field being used, so I'd be inclined to remove it for now. I know that we might want to expose the difference here via the reflection API, but I'd rather not do so in a way that relies on storing extra information in the AST nodes that reference these types.

(The way that texture/resource types use the special extra "payload" data field for magic types is a gross hack that I am not happy with at all. I'd like to see us trend in the direction of using it less rather than using it for more types.)